onnxruntime/core/graph/contrib_ops/bert_defs.cc

```diff
@@ -302,6 +309,94 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
           AttentionTypeAndShapeInference(ctx, past_input_index);
         }));
 
+constexpr const char* DecoderMaskedMultiheadAttention_ver1_doc = R"DOC(
+Uni-directional attention that supports input sequence length of 1.
+
+The weights for input projection of Q, K and V are merged. The data is stacked on the second dimension. Its shape
+is (input_hidden_size, hidden_size + hidden_size + v_hidden_size). Here hidden_size is the hidden dimension of Q and K,
+and v_hidden_size is that of V.
+
+The mask_index is optional. If it is provided, only raw attention mask with shape (batch_size, total_sequence_length) is supported currently.
+
+Both past and present state need to be provided.
+
+The qkv_hidden_sizes is required only when K and V have different hidden sizes.
+
+The total_sequence_length is past_sequence_length + kv_sequence_length. Here kv_sequence_length is the length of K or V.
+Currently, only self attention is supported which means that kv_sequence_length equals to sequence_length (sequence length of Q).
+)DOC";
+
+ONNX_MS_OPERATOR_SET_SCHEMA(
+    DecoderMaskedMultiheadAttention, 1,
+    OpSchema()
+        .SetDoc(DecoderMaskedMultiheadAttention_ver1_doc)
+        .Attr("num_heads", "Number of attention heads", AttributeProto::INT)
+        .Attr("past_present_share_buffer",
+              "Corresponding past and present are same tensor, its size is "
+              "(2, batch_size, num_heads, max_sequence_length, head_size)",
+              AttributeProto::INT,
+              OPTIONAL_VALUE)
+        .Attr("scale",
+              "Custom scale will be used if specified. Default value is 1/sqrt(head_size)",
+              AttributeProto::FLOAT,
+              OPTIONAL_VALUE)
+        .Input(0,
+               "input",
+               "Input tensor with shape (batch_size, 1, input_hidden_size)",
+               "T")
+        .Input(1,
+               "weights",
+               "Merged Q/K/V weights with shape (input_hidden_size, hidden_size + hidden_size + v_hidden_size)",
+               "T")
+        .Input(2,
+               "bias",
+               "Bias tensor with shape (hidden_size + hidden_size + v_hidden_size) for input projection",
+               "T")
+        .Input(3,
+               "mask_index",
+               "Mask values of shape (batch_size, total_sequence_length)",
+               "M",
+               OpSchema::Optional)
+        .Input(4,
```

Comment:
I saw the cached past is reordered to [B, N, head_size / x, max_length, x], do we still accept the (2, batch_size, num_heads, past_sequence_length, head_size) here?